bftengine/src/bcstatetransfer/SourceSelector.hpp

```diff
@@ -0,0 +1,80 @@
+// Concord
+//
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+#pragma once
+
+#include <random>
+#include <set>
+#include <stdint.h>
+#include <sstream>
+
+#include "assertUtils.hpp"
+
+namespace bftEngine {
+namespace SimpleBlockchainStateTransfer {
+namespace impl {
+
+static const uint16_t NO_REPLICA = UINT16_MAX;
+
+// Information about which current source is selected and which replicas are
+// preferred, as well as data that helps to select a current source replica.
+class SourceSelector {
+ public:
+  std::set<uint16_t> preferredReplicas_;
+  uint16_t currentReplica_ = NO_REPLICA;
+
+ private:
+  uint64_t sourceSelectionTimeMilli_ = 0;
+  std::set<uint16_t> allOtherReplicas_;
```

Comment:
What does `Other` refer to? Other as in "not a source", or as in "not preferred", or as in "not current"?

Reply:
It refers to replicas that are not this replica. I'm all ears for a better name :)